### PR TITLE
Use signatures in OpenQA::Shared::Plugin::Gru

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Review by Martchus — APPROVED** [review]

**Comment by okurz** [comment]
codecov/patch failed, I would go ahead anyway

**Review by baierjan — APPROVED** [review]

File: lib/OpenQA/Schema.pm [hunk 1]
```diff
@@ -100,19 +87,16 @@ sub deploy {
 }
 
 # Class attribute used for testing with OpenQA::Test::Database
-sub search_path_for_tests {
-    my $class = shift;
+sub search_path_for_tests ($class, $new_search_path = undef) {
     state $search_path;
-    $search_path = shift if @_;
+    $search_path = $new_search_path if defined $new_search_path;
```
Comment (perlpunk) [inline lib/OpenQA/Schema.pm:90]: There's no way to set `$search_path` to undef now.

Comment (okurz) [reply lib/OpenQA/Schema.pm:90]: thx, I have reverted in the meantime and will look into doing this properly again

Comment (okurz) [reply lib/OpenQA/Schema.pm:90]: What do you suggest we do here instead?

File: lib/OpenQA/Shared/Plugin/Gru.pm [hunk 2]
```diff
@@ -47,13 +47,14 @@ sub register {
     my $schema = $app->schema;
 
     my $conn = Mojo::Pg->new;
-    if (ref $schema->storage->connect_info->[0] eq 'HASH') {
-        $self->dsn($schema->dsn);
-        $conn->username($schema->storage->connect_info->[0]->{user});
-        $conn->password($schema->storage->connect_info->[0]->{password});
+    my $connect_info = $schema->storage->connect_info->[0];
+    if (ref $connect_info eq 'HASH') {
+        $self->dsn($connect_info->dsn);
```
Comment (perlpunk) [inline lib/OpenQA/Shared/Plugin/Gru.pm:47]:
```suggestion
$self->dsn($schema->dsn);
```

File: lib/OpenQA/Schema.pm (hunk 1 above)
Comment (perlpunk) [reply lib/OpenQA/Schema.pm:90]:
```perl
sub search_path_for_tests ($class, @args) {
    state $search_path;
    $search_path = shift @args if @args;
```

File: lib/OpenQA/Shared/Plugin/Gru.pm (hunk 2 above)
Comment (okurz) [reply lib/OpenQA/Shared/Plugin/Gru.pm:47]: applied in #4928

File: lib/OpenQA/Schema.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/Schema.pm:90]: applied in
* #4927

**Pull request merged**